src/main/java/core/basesyntax/Application.java

```diff
@@ -2,6 +2,11 @@
 
 public class Application {
     public static void main(String[] args) {
-        // create three balls using class Lottery and print information about them in console
+        Lottery lottery = new Lottery(new Ball(),
+                new ColorSupplier(),
+                new NumberBall());
```

Comment:
It seems you are passing new instances of Ball, ColorSupplier, and NumberBall to the Lottery constructor. This might not be necessary if Lottery can create these instances internally. If Lottery requires external instances, consider using a single instance of ColorSupplier and NumberBall if they do not maintain state that changes during the loop iterations.